openstack/sharedfilesystems/v2/sharetypes/requests.go

```diff
@@ -113,7 +113,7 @@ type SetExtraSpecsOptsBuilder interface {
 
 type SetExtraSpecsOpts struct {
 	// A list of all extra specifications to be added to a ShareType
-	Specs map[string]interface{} `json:"extra_specs"`
+	ExtraSpecs ExtraSpecsOpts `json:"extra_specs" required:"true"`
```

Comment:
This should stay `map[string]interface{}`. According to the [Manila documentation](https://docs.openstack.org/manila/latest/contributor/capabilities_and_extra_specs.html), `driver_handles_share_servers` and `snapshot_support` are required for all drivers, which is most likely why they are part of `CreateOpts`.

But Manila can support other types of extra specs, depending on the driver in use. Since we'll never know the full list of supported options, `map[string]interface{}` can be used so users can add any extra spec they need.

OpenStack can be confusing like this. I would have done the same thing myself. I have a test environment that runs Manila and I noticed I could add other key/values, so that's what confirmed `map[string]interface{}`.

`required:"true"`: this is a good catch. It turns out the API will trigger an error if `ExtraSpecs` is left empty :)

Reply:
I will make the necessary changes. Thanks